### ISSUE TITLE
Allow custom naming of the Add new button for each form. e.g. Add new file etc.

### ISSUE THREAD

**Issue by harmeet-status** [sartography/spiff-arena#1272]

**Comment by harmeet-status** [comment]
@sashayar13 do we already have this capability?

**Comment by sashayar13** [comment]
no, we don't

**Comment by burnettk** [comment]
we actually do! maybe we should make this a documentation ticket. this is what a uischema would look like:

```json
{
  "firstName": {
    "ui:autofocus": true,
    "ui:description": "Make text **bold** or *italic*. Take a look at other options [here](https://probablyup.com/markdown-to-jsx/)."
  },
  "ui:submitButtonOptions": {
    "submitText": "Make it so"
  }
}
```

**Comment by jasquat** [comment]
I think this may be related to the add new button when using an array. I believe the submitButtonOptions is for the form submit button.

For example:
```
{
  "title": "Nested Form / Repeating Section",
  "description": "Allow the form submitter to add multiple entries for a set of fields.",
  "type": "object",
  "properties": {
    "tasks": {
      "type": "array",
      "title": "Tasks",
      "items": {
        "type": "object",
        "required": [
          "title"
        ],
        "properties": {
          "title": {
            "type": "string",
            "title": "Title",
            "description": "Please describe the task to complete"
          },
          "done": {
            "type": "boolean",
            "title": "Done?",
            "default": false
          }
        }
      }
    }
  }
}
```

We may be able to change this in https://github.com/sartography/spiff-arena/blob/main/spiffworkflow-frontend/src/rjsf/carbon_theme/AddButton/AddButton.tsx#L19

**Comment by burnettk** [comment]
thank you, totally misinterpreted this, and thought it was the main submit for the form.

**Comment by harmeet-status** [comment]
Guys to reduce confusion, these are the buttons that I'm talking about.
<img width="675" alt="Screenshot 2024-03-27 at 11 17 33 AM" src="https://github.com/sartography/spiff-arena/assets/104047049/0c9dfcac-56bb-49ce-8fa6-5681735610d0">

**Comment by usama9500** [comment]
Documented Nested Forms in this PR: https://github.com/sartography/spiff-arena/pull/1282

**Comment by harmeet-status** [comment]
> I think this may be related to the add new button when using an array. I believe the submitButtonOptions is for the form submit button.
> 
> For example:
> 
> ```json
> {
>   "title": "Nested Form / Repeating Section",
>   "description": "Allow the form submitter to add multiple entries for a set of fields.",
>   "type": "object",
>   "properties": {
>     "tasks": {
>       "type": "array",
>       "title": "Tasks",
>       "items": {
>         "type": "object",
>         "required": [
>           "title"
>         ],
>         "properties": {
>           "title": {
>             "type": "string",
>             "title": "Title",
>             "description": "Please describe the task to complete"
>           },
>           "done": {
>             "type": "boolean",
>             "title": "Done?",
>             "default": false
>           }
>         }
>       }
>     }
>   }
> }
> ```
> 
> We may be able to change this in https://github.com/sartography/spiff-arena/blob/main/spiffworkflow-frontend/src/rjsf/carbon_theme/AddButton/AddButton.tsx#L19

Does making this change once allow all the buttons to have custom names?

**Comment by harmeet-status** [comment]
> It looks like we could make it per form instead of globally.
> 
> Although it looks like rjsf by default doesn't allow text on that button at all and instead is only `+` with a title (tooltip) that can only be overridden globally. This means when we switch to mui we'll still have to override the Addbutton component to keep it as is.
> 
> Is the default plus sign only unacceptable?

Yeah we've had people tell us that + is confusing, it should state what are you adding

**Comment by harmeet-status** [comment]
This ticket is also linked to https://github.com/sartography/spiff-arena/issues/1697 where we are testing how the new MUI 5 looks for the existing forms.
